server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -198,7 +202,16 @@ private boolean prepareAndUpload(Long primaryTerm, Long generation) throws IOExc
     }
 
     private boolean upload(Long primaryTerm, Long generation) throws IOException {
-        logger.trace("uploading translog for {} {} ", primaryTerm, generation);
+        boolean primaryMode = primaryModeSupplier.getAsBoolean();
+        // During primary relocation (primary-primary peer recovery), both the old and the new primary have engine
+        // created with the RemoteFsTranslog. Both primaries are equipped to upload the translogs. The primary mode check
+        // below ensures that the real primary only is uploading.
+        if (primaryMode == false) {
+            logger.trace("skipped uploading translog for {} {}", primaryTerm, generation);
+            // NO-OP
+            return true;
```

Comment:
Rather than `upload`, we should bail out before we sync to local as well .

Reply:
Currently the upload is coupled with multiple usecases as mentioned in https://github.com/opensearch-project/OpenSearch/issues/5906. I had tried the changed suggested by you. Have reverted the changes specifying more details in the issue created.